static/js/message_events.js

```diff
@@ -177,6 +185,21 @@ exports.update_messages = function update_messages(events) {
                         }
                     }
                 }
+
+                if (current_filter && stream_changed &&
+                    current_filter.has_topic(stream_name, orig_topic)) {
+                    const new_filter = current_filter.filter_with_new_params({
+                        operator: 'stream',
+                        operand: new_stream_name,
+                    });
+                    const operators = new_filter.operators();
+                    const opts = {
+                        trigger: 'stream change',
+                        then_select_id: current_id,
+                    };
+                    narrow.activate(operators, opts);
+                    changed_narrow = true;
```

Comment:
I think this block should be combined with the previous one -- we want to only renarrow once if we later add stream+topic edits in a single event.  And we want the `selection_changed_topic` conditional (which I guess should be rebranded as `seleted_message_moved`)